examples/trials/mnist-cascading-search-space/search_space.json

```diff
@@ -1,62 +1,114 @@
-		["Max_pool", {"_type":"choice","_value":[2,3,5]}],
-		["Avg_pool", {"_type":"choice","_value":[2,3,5]}]
-	]},
-   "layer5":{"_type":"choice","_value":[
-		"Empty",
-		["Conv", {"_type":"choice","_value":[2,3,5]}],
-		["Max_pool", {"_type":"choice","_value":[2,3,5]}],
-		["Avg_pool", {"_type":"choice","_value":[2,3,5]}]
-	]},
-   "layer6":{"_type":"choice","_value":[
-		"Empty",
-		["Conv", {"_type":"choice","_value":[2,3,5]}],
-		["Max_pool", {"_type":"choice","_value":[2,3,5]}],
-		["Avg_pool", {"_type":"choice","_value":[2,3,5]}]
-	]},
-   "layer7":{"_type":"choice","_value":[
-		"Empty",
-		["Conv", {"_type":"choice","_value":[2,3,5]}],
-		["Max_pool", {"_type":"choice","_value":[2,3,5]}],
-		["Avg_pool", {"_type":"choice","_value":[2,3,5]}]
-	]},
-   "layer8":{"_type":"choice","_value":[
-		"Empty",
-		["Conv", {"_type":"choice","_value":[2,3,5]}],
-		["Max_pool", {"_type":"choice","_value":[2,3,5]}],
-		["Avg_pool", {"_type":"choice","_value":[2,3,5]}]
-	]},
-   "layer9":{"_type":"choice","_value":[
-		"Empty",
-		["Conv", {"_type":"choice","_value":[2,3,5]}],
-		["Max_pool", {"_type":"choice","_value":[2,3,5]}],
-		["Avg_pool", {"_type":"choice","_value":[2,3,5]}]
-	]}
+	"layer0": {
+		"_type": "choice",
+		"_value": [{
+				"_name": "Empty"
+			},
+			{
+				"_name": "Conv",
+				"kernel_size": {
+					"_type": "choice",
+					"_value": [1, 2, 3, 5]
+				}
+			},
+			{
+				"_name": "Max_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			},
+			{
+				"_name": "Avg_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			}
+		]
+	},
+	"layer1": {
+		"_type": "choice",
+		"_value": [{
+				"_name": "Empty"
+			},
+			{
+				"_name": "Conv",
+				"kernel_size": {
+					"_type": "choice",
+					"_value": [1, 2, 3, 5]
+				}
+			},
+			{
+				"_name": "Max_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			},
+			{
+				"_name": "Avg_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			}
+		]
+	},
+	"layer2": {
+		"_type": "choice",
+		"_value": [{
+				"_name": "Empty"
+			},
+			{
+				"_name": "Conv",
+				"kernel_size": {
+					"_type": "choice",
+					"_value": [1, 2, 3, 5]
+				}
+			},
+			{
+				"_name": "Max_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			},
+			{
+				"_name": "Avg_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			}
+		]
+	},
+	"layer3": {
+		"_type": "choice",
+		"_value": [{
+				"_name": "Empty"
+			},
+			{
+				"_name": "Conv",
+				"kernel_size": {
+					"_type": "choice",
+					"_value": [1, 2, 3, 5]
+				}
+			},
+			{
+				"_name": "Max_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]
+				}
+			},
+			{
+				"_name": "Avg_pool",
+				"pooling_size": {
+					"_type": "choice",
+					"_value": [2, 3, 5]

```

Comment:
Can you add an example with **three levels** of nesting?

Reply:
I don't think we need complex examples.

Reply:
I test **three levels** of nesting in local successfully but I also don't want to put a complex example here because it would confuse users sometimes, I think.

Reply:
But it also confuses me... If I have three levels of nested search space, I have to use "_name" on the third level or not?